doc/Dump-Utility.md

```diff
@@ -0,0 +1,403 @@
+# Dump Utility for Easy Debugging #
+#### Rev 0.1
+
+# Table of Contents
+  * [Revision](#revision)
+  * [About this Manual](#about-this-manual)
+  * [Definitions/Abbreviation](#definitionsabbreviations)
+  * [Overview](#overview)
+  * [1. Requirements](#1-requirements)
+      * [1.1 Intended Usage](#11-intended-usage)
+      * [1.2 Customization Options](#12-customization-options)
+      * [1.3 Extensibility](#13-extensibility)
+      * [1.4 Directory Structure](#14-directory-structure)
+  * [2. Design](#2-design)
+      * [2.0 The Executor Class](#20-the-executor-class)
+      * [2.1 Adding new Modules](#21-adding-new-modules)
+      * [2.2 Module Internals](#22-module-internals)
+      * [2.3 Helper Methods Available](#23-helper-methods-available)
+      * [2.4 Match Infrastructure](#24-Match-Infrastructure)
+      * [2.5 MatchRequest Examples](#25-matchrequest-examples)
+  * [3. Unit Tests](#3-unit-tests)
+  * [4. TechSupport](#4-techsupport)
+   
+### Revision  
+
+| Rev |     Date    |       Author       | Change Description          |
+|:---:|:-----------:|:-------------------------|:----------------------|
+| 0.1 | 05/01/2021  | Vivek Reddy Karri        | Initial version       |
+
+## About this Manual
+This document describes the details of a dump cli utility which collects and dumps the redis state for a given feature/module. 
+
+## Definitions/Abbreviations 
+###### Table 1: Abbreviations
+| Abbreviation | Description                                                  |
+| ------------ | ------------------------------------------------------------ |
+| APP DB       | Application Database                                         |
+| ASIC DB      | ASIC Database                                                |
+| CONF DB      | Configuration Database                                       |
+
+
+                     
+In this document, the term '**redis state**' refers to the intermediate state of a given feature present across all the Redis DB's 
+
+## Overview 
+In SONiC, there usually exists a set of tables related/relevant to a particular module. All of these have to be looked at to confirm whether  any configuration update is properly applied and propagated.
+
+The task of debugging quickly becomes tedious because currently, there is no utility which does print a unified view of the redis-state.
+This is the problem which is addressed by this dump utility.
+
+This utility provides the base infrastructure and guidelines to make is easy for the developers to extend and support the utility for different modules.
+
+## **1 Requirements**
+
+### 1.1 Intended Usage
+
+Note: Currently, `state` is the only subcommand under `dump`.
+
+```
+dump state <feature/module> <arg>
```

Comment:
@vivekreddynv the description seems to be invalid: `dump state <feature/module> <args | all>`

Reply:
Done!